internal/nginx/main.go

```diff
@@ -128,15 +134,24 @@ func ReadFileToString(path string) (string, error) {
 	return string(contents), nil
 }
 
-func buildUnixSocketClient() *http.Client {
+func buildUnixSocketClient(timeout time.Duration) *http.Client {
 	u := &httpunix.Transport{
 		DialTimeout:           1 * time.Second,
-		RequestTimeout:        10 * time.Second,
-		ResponseHeaderTimeout: 10 * time.Second,
+		RequestTimeout:        timeout * time.Second,
+		ResponseHeaderTimeout: timeout * time.Second,
 	}
 	u.RegisterLocation(statusLocation, StatusSocket)
 
 	return &http.Client{
 		Transport: u,
 	}
 }
+
+func getOrSetUnixSocketClient() *http.Client {
```

Comment:
Please remove this and just use `buildUnixSocketClient(HealthCheckTimeout).Post(.....`

Reply:
why? the `socketClient` is gloabl variable in the previous, call `buildUnixSocketClient ` at each request would initial one client per request, it costs memory which is valueless.

Reply:
Okay, things cleared, and I will appreciate if you share with me the reason about code in this way :)   @aledbf